apps/hash-frontend/src/pages/[shortname].page/edit-pinned-entity-types-modal.tsx

```diff
@@ -0,0 +1,424 @@
+      },
+    });
+
+    /** @todo: error handling */
+
+    setLoading(false);
+
+    void refetchProfile();
+
+    reset({ pinnedEntityTypes });
+    onClose();
+  });
+
+  const handleOnDragEnd = useCallback<OnDragEndResponder>(
+    (result) => {
+      if (!result.destination) {
+        return;
+      }
+
+      const items = Array.from(fields);
+      const [reorderedItem] = items.splice(result.source.index, 1);
+
+      if (!reorderedItem) {
+        return;
+      }
+
+      items.splice(result.destination.index, 0, reorderedItem);
+
+      replace(items);
+    },
+    [fields, replace],
+  );
+
+  const handleCreateNewEntityType = useCallback(
+    async (title: string) => {
+      const { data: createdEntityType } = await createEntityType({
+        data: {
+          entityType: {
+            title,
+            type: "object",
+            properties: {},
+          },
+        },
+      });
+
+      if (createdEntityType) {
+        append(createdEntityType);
+      } else {
+        /** @todo: error handling */
+      }
+      setDisplayEntityTypesSearch(false);
+    },
+    [createEntityType, append],
+  );
+
+  const handleDiscard = useCallback(() => {
+    reset();
+    onClose();
+  }, [reset, onClose]);
+
+  const renderDraggable = useDraggableInPortal();
+
+  const isSubmitDisabled = Object.keys(formState.dirtyFields).length === 0;
+
+  return (
+    <Modal
+      {...modalProps}
+      sx={{
+        "> div": {
+          overflow: "hidden",
+          padding: 0,
+        },
+      }}
+      onClose={onClose}
+    >
+      <Box>
+        <Box
+          paddingX={3}
+          paddingBottom={1.5}
+          paddingTop={2}
+          display="flex"
+          justifyContent="space-between"
+        >
+          <Box>
+            <Typography
+              gutterBottom
+              sx={{
+                fontSize: 16,
+                fontWeight: 500,
+                color: ({ palette }) => palette.gray[80],
+              }}
+            >
+              Pinned types
+            </Typography>
+            <Typography
+              sx={{ fontSize: 14, color: ({ palette }) => palette.gray[80] }}
+            >
+              Choose up to 5 types to appear in the top-bar of your profile
+            </Typography>
+          </Box>
+          <Box>
+            <IconButton
+              onClick={onClose}
+              sx={{ marginRight: -2, marginTop: -1 }}
+            >
+              <XMarkRegularIcon />
+            </IconButton>
+          </Box>
+        </Box>
+        <Divider />
+        <Box
+          component="form"
+          onSubmit={innerSubmit}
+          padding={3}
+          paddingTop={2.25}
+        >
+          <Typography
+            variant="smallCaps"
+            sx={{ color: ({ palette }) => palette.gray[70] }}
+          >
+            Types
+          </Typography>
+          <DragDropContext onDragEnd={handleOnDragEnd}>
+            <Droppable droppableId="pinnedEntityTypes">
```

Comment:
If possible I think it would be better to lock the drag area to this element so that users can't drag the item all around the screen.

Reply:
I spent about 15 minutes trying to get this to work using the `getContainerForClone` prop, and reworking the custom `useDraggableInPortal` hook that I had to add, but couldn't get something working. I think it'd be a good idea, but not sure it's worth investing more time - I can create a follow-up ticket?

Reply:
I wouldn't worry about a follow up, we can change it if it becomes annoying.

Out of interest why the new dnd library? `dnd-kit` handles locking the draggable area / axes with a couple of props.

Reply:
To be honest mainly because I had used it before, and I thought `@dnd-kit` didn't support the animations when rearranging but having taken another look at the docs I was wrong about that. It may have been a better choice, would you like me to replace it? Or do it in follow-up?

Reply:
Have created the ticket: https://linear.app/hash/issue/H-1013/replace-react-beautiful-dnd-with-dnd-kit-in-editpinnedentitytypesmodal

Reply:
Thank you